python/plugins/grassprovider/ext/r_horizon.py

```diff
@@ -38,14 +39,42 @@ def checkParameterValuesBeforeExecuting(alg, parameters, context):
 
 
 def processOutputs(alg, parameters, context, feedback):
+    # Inspired from GRASS implementation
+    def getNumberDecimals(number):
+        """ Return the number of decimals of a number (int or float) """
+        if int(number) == number:
+            return 0
+        return len(str(number).split(".")[-1])
+
+    def doubleToBaseName(number, nDecimals):
+        """
+        Format filename, according to GRASS implementation,
+        based on provided number and number of decimals
+        """
+        number += 0.0001
+        if nDecimals == 0:
+            return f'{int(number):03}'
+        int_part = int(number)
+        dec_part = int((number - int_part) * pow(10, nDecimals))
```

Comment:
I would use a better conversion due to floating-point errors (modf, Decimal + str, or something like).
But, maybe your hack later is sufficient?

Note for later: a convenient method could be useful and reusable elsewhere.

Reply:
...and this replicates the logic in https://github.com/OSGeo/grass/blob/6253da1bd6ce48d23419e99e8b503edf46178490/lib/gis/basename.c#L97-L101
in order to ensure that the values calculated (also replicating the same rounding errors) by the GRASS module and used to generate the output files names are exactly the same of the ones calculated by QGIS.
AFAIK, GRASS uses this logic only in the r.horizon module.

Reply:
OK, thanks!